aten/src/ATen/native/cuda/Indexing.cu

```diff
@@ -245,7 +247,325 @@ void index_put_accum_kernel(Tensor & self, TensorList indices, const Tensor & va
 
 REGISTER_CUDA_DISPATCH(index_put_accum_stub, &index_put_accum_kernel);
 } //anonymous
-} //at
-} //native
 
 
+// Check tensor dimensions for index operations, and return the slice size.
+static ptrdiff_t getSliceSize(const Tensor & dst,
+                              int dim,
+                              const Tensor & index,
+                              const Tensor & src)
+{
+  int dstDims = dst.dim();
+  int srcDims = src.dim();
+
+  TORCH_CHECK(index.dim() <= 1, "Index must be vector or scalar");
+
+  ptrdiff_t dstSliceSize = 1;
+  TORCH_CHECK(dim >= 0 && dim < dstDims, "Indexing dim ", dim, " is out of bounds");
+  for (int d = 0; d < dstDims; d++) {
+    if (d != dim) {
+      dstSliceSize *= dst.size(d);
+    }
+  }
+
+  TORCH_CHECK(dim < srcDims, "Indexing dim ", dim, " is out of bounds");
+  TORCH_CHECK(index.numel() == src.size(dim),
+             "length of src.size[dim] is not equal to length of indices");
+
+  ptrdiff_t srcSliceSize = 1;
+  bool mismatch = false;
+
+  if (dstDims != srcDims) mismatch = true;
+
+  for (int d = 0; d < srcDims; d++) {
+    if (d != dim) {
+      srcSliceSize *= src.size(d);
+      if (!mismatch && dst.size(d) != src.size(d)) mismatch = true;
+    }
+  }
+
+  TORCH_CHECK(dstSliceSize == srcSliceSize,
+             "Source/destination tensor have different slice sizes (%ld vs %ld)",
+             dstSliceSize, srcSliceSize);
+
+  if (mismatch) {
+    static bool warningShown = false;
+    if (!warningShown) {
```

Comment:
Please use `TORCH_WARN_ONCE` instead of this construction.

Reply:
fixed.